core/src/components/input/input.tsx

```diff
@@ -240,7 +240,10 @@ export class Input implements ComponentInterface {
        * Used for patterns such as input trimming (removing whitespace),
        * or input masking.
        */
-      this.nativeInput.value = this.getValue();
+       const { selectionStart, selectionEnd } = this.nativeInput;
+       this.nativeInput.value = this.getValue();
+       // Set the cursor position back to where it was before the value change
+       this.nativeInput.setSelectionRange(selectionStart, selectionEnd);
```

Comment:
Can you make a tech debt ticket to remove this when we drop support for iOS 15.3? The issue may have appeared as a result of one of your other changes, but the behavior is definitely a bug in iOS given that it reproduces in 15.3 but not in 15.4.

Reply:
Yes, can do!